pkg/utils/build/quay.go

```diff
@@ -0,0 +1,60 @@
+package build
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	. "github.com/redhat-appstudio/image-controller/pkg/quay"
+)
+
+var (
+	quayApiUrl = "https://quay.io/api/v1"
+	quayOrg    = utils.GetEnv("DEFAULT_QUAY_ORG", "redhat-appstudio-qe")
+	quayToken  = utils.GetEnv("DEFAULT_QUAY_ORG_TOKEN", "")
+	quayClient = NewQuayClient(&http.Client{Transport: &http.Transport{}}, quayToken, quayApiUrl)
+)
+
+func GetQuayImageName(annotations map[string]string) (string, error) {
+	type imageAnnotation struct {
+		Image  string `json:"Image"`
+		Secret string `json:"Secret"`
+	}
+	image_annotation_str := annotations["image.redhat.com/image"]
+	var ia imageAnnotation
+	err := json.Unmarshal([]byte(image_annotation_str), &ia)
+	if err != nil {
+		return "", err
+	}
+	tokens := strings.Split(ia.Image, "/")
+	return strings.Join(tokens[2:], "/"), nil
+}
+
+func GetRobotAccountName(imageName string) string {
+	tokens := strings.Split(imageName, "/")
+	return strings.Join(tokens, "")
+}
+
+func DoesImageRepoExistInQuay(quayImageRepoName string) (bool, error) {
+	exists, err := quayClient.DoesRepositoryExist(quayOrg, quayImageRepoName)
+	if exists && err == nil {
```

Comment:
imo this is not needed. [if the image exists, err is always nil](https://github.com/redhat-appstudio/image-controller/blob/e2e987ef8f06fd48141055f2ed8ce0b93c5dbb16/pkg/quay/quay.go#L119)

Reply:
Good catch. Changed now.